Makefile

```diff
@@ -14,37 +14,46 @@ else
 override O := $(BUILDDIR)/$(O)
 endif
 
-.NOTPARALLEL: $(TARGETS) $(TARGETS_CONFIG) all
+################################################################################
 
-.PHONY: $(TARGETS) $(TARGETS_CONFIG) all buildroot-help help
+COLOR_STEP := $(shell tput smso 2>/dev/null)
+COLOR_WARN := $(shell (tput setab 3; tput setaf 0) 2>/dev/null)
+TERM_RESET := $(shell tput sgr0 2>/dev/null)
 
-all: $(TARGETS)
+################################################################################
+
+.NOTPARALLEL: $(TARGETS) $(TARGETS_CONFIG) default
+
+.PHONY: $(TARGETS) $(TARGETS_CONFIG) default buildroot-help help
+
+# fallback target when target undefined here is given
+.DEFAULT:
+	@echo "$(COLOR_STEP)=== Falling back to Buildroot target '$@' ===$(TERM_RESET)"
+	$(MAKE) -C $(BUILDROOT) O=$(O) BR2_EXTERNAL=$(BUILDROOT_EXTERNAL) "$@"
+
+# default target when no target is given - must be first in Makefile
+default:
+	$(MAKE) -C $(BUILDROOT) O=$(O) BR2_EXTERNAL=$(BUILDROOT_EXTERNAL)
 
 $(TARGETS_CONFIG): %-config:
-	@echo "config $*"
+	@if [ -d $(O) ] && [ "$$(ls -A $(O))" ]; then \
+		echo "$(COLOR_WARN)WARNING: Output directory '$(O)' is not empty!$(TERM_RESET)"; \
+		echo "         This may cause errors if it was previously used for a different target."; \
+		echo "         Before running build for a different target, run 'make distclean' first."; \
+		echo ""; \
```

Comment:
Can we maybe check what target was used for a given output directory before? :thinking: 

With this I'll get a warning for every build, I typically use command like this:
```
./scripts/enter.sh make O=output_ova ova
```

Reply:
I thought about it as a warning for newcomers (as you can also do e.g. `make ova_defconfig && make`) so if you knew what you were doing, you could ignore it. But yeah, why not. Adjusted the condition to check the `.config` before regenerating it.

Reply:
Nice, yeah that seems to work here.

One more nit: Can we maybe add a sleep of 10s or so and continue only then? Right now if you see that error it essentially is "too late", your build directory is already tainted with that other defconfig :thinking:

Reply:
Sure, done as well.